src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets

```diff
@@ -304,7 +304,7 @@ Copyright (c) .NET Foundation. All rights reserved.
       </PropertyGroup>
     </When>
     
-    <When Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(OutputType)' == 'Exe'">
+    <When Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(IsExecutable)' == 'true'">
```

Comment:
I was under the assumption that things like Test projects were going to set `$(IsExecutable)` to true.  If that's the case, then I don't think these two lines should be changed.

The "Run Information" is used during F5 and `dotnet run`.  If you try to `dotnet run` a Test project, will it try to execute the Test's $(TargetPath), which is a .dll on .NET Framework.  Thus failing with a bad error.